CMakeLists.txt

```diff
@@ -555,6 +555,12 @@ if(AVIF_BUILD_TESTS)
     endif()
 endif()
 
+if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
+    set(CMAKE_INSTALL_LIBDIR "lib")
+endif()
+if(NOT DEFINED CMAKE_INSTALL_INCLUDEDIR)
+    set(CMAKE_INSTALL_INCLUDEDIR "include")
+endif()
```

Comment:
(Please note my question at the end of this comment.)

Earlier in this file at lines 456-458, we have:

```
if(NOT SKIP_INSTALL_ALL)
    include(GNUInstallDirs)
endif()
```

The [CMake documentation for GNUInstallDirs](https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html) says the GNUInstallDirs module defines the `CMAKE_INSTALL_LIBDIR` and `CMAKE_INSTALL_INCLUDEDIR` variables. Therefore, it seems that it would be better to provide the default values of these two variables by rewriting lines 456-458 as follows:

```
if(NOT SKIP_INSTALL_ALL)
    include(GNUInstallDirs)
else()
    set(CMAKE_INSTALL_LIBDIR "lib")
    set(CMAKE_INSTALL_INCLUDEDIR "include")
endif()
```

**IMPORTANT:** I verified that this file uses `CMAKE_INSTALL_LIBDIR` and `CMAKE_INSTALL_INCLUDEDIR` only if `NOT SKIP_INSTALL_ALL`. Does this mean the GNUInstallDirs module sometimes does not define these two variables?

Reply:
From my understanding the `configure_file(libavif.pc.cmake ...` is run regardless of `SKIP_INSTALL_ALL` so that's why a fallback was added for when `GNUInstallDirs` was not included. I'll change to above if you feel it is more readable...

Also, any reason not to include(GNUInstallDirs) always so we don't have to do this?

Alternatively, any reason not to run `configure_file(libavif.pc.cmake ...` only if `if(NOT SKIP_INSTALL_ALL)` as well?

Reply:
Hi Miloš,

Thank you for the reply. Now that I understand the bug a little better, I probably would suggest running `configure_file(libavif.pc.cmake ...` only if `NOT SKIP_INSTALL_ALL`.

Since I am not familiar with this part of the build system, could you explain how to reproduce this bug so that I can take a closer look?

Reply:
Yes, this is what I did in the last commit, now `configure` of the .pc file is right next to the `install` of the .pc file.

Without this PR, you should still see the placeholder `libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@` inside the produced .pc file when `-DCMAKE_SKIP_INSTALL_RULES` is used for cmake configuration.

Reply:
> Without this PR, you should still see the placeholder `libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@` inside the produced .pc file when `-DCMAKE_SKIP_INSTALL_RULES` is used for cmake configuration.

If I pass `-DCMAKE_SKIP_INSTALL_RULES=ON` to cmake, cmake emits a warning:

```
CMake Warning in CMakeLists.txt:
  CMAKE_SKIP_INSTALL_RULES was enabled even though installation rules have
  been specified
```

And the produced libavif.pc file has the line

`libdir=${prefix}/`

rather than the line you described:

`libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@`

I guess you meant @CMAKE_INSTALL_LIBDIR@ has been substituted with an empty string?

Reply:
Seems to be the case indeed... incorrect/undesired anyway.